pip_audit/_dependency_source/requirement.py

```diff
@@ -198,7 +198,10 @@ def _fix_file(self, filename: Path, fix_version: ResolvedFixVersion) -> None:
         with filename.open("w") as f:
             found = False
             for req in reqs:
-                if isinstance(req, InstallRequirement) and req.name == fix_version.dep.name:
+                if (
+                    isinstance(req, InstallRequirement)
+                    and req.name == fix_version.dep.canonical_name
```

Comment:
Does `req.name` also need to be canonicalized here?

Reply:
[75fa6d0](https://github.com/pypa/pip-audit/pull/577/commits/75fa6d0d22294c548496b44d578a47265573fcd7)

Yep, good catch.